src/test/javascript/spec/component/admin/health.component.spec.ts

```diff
@@ -6,61 +6,86 @@ import { ArtemisTestModule } from '../../test.module';
+import { TranslatePipeMock } from '../../helpers/mocks/service/mock-translate.service';
+import { JhiConnectionStatusComponent } from 'app/shared/connection-status/connection-status.component';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
 
 describe('HealthComponent', () => {
     let comp: HealthComponent;
     let fixture: ComponentFixture<HealthComponent>;
-    let service: HealthService;
+    let healthService: HealthService;
+    let modalService: NgbModal;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
             imports: [ArtemisTestModule],
-            declarations: [HealthComponent],
-        })
-            .overrideTemplate(HealthComponent, '')
-            .compileComponents();
+            declarations: [HealthComponent, MockComponent(HealthModalComponent), TranslatePipeMock, MockComponent(JhiConnectionStatusComponent), MockDirective(TranslateDirective)],
+            providers: [{ provide: NgbModal, useClass: MockNgbModalService }],
+        }).compileComponents();
     });
 
     beforeEach(() => {
         fixture = TestBed.createComponent(HealthComponent);
         comp = fixture.componentInstance;
-        service = fixture.debugElement.injector.get(HealthService);
+        healthService = TestBed.inject(HealthService);
+        modalService = TestBed.inject(NgbModal);
     });
 
-    describe('getBadgeClass', () => {
-        it('should get badge class', () => {
-            const upBadgeClass = comp.getBadgeClass('UP');
-            const downBadgeClass = comp.getBadgeClass('DOWN');
-            expect(upBadgeClass).toEqual('bg-success');
-            expect(downBadgeClass).toEqual('bg-danger');
-        });
+    it('should get badge class', () => {
+        const upBadgeClass = comp.getBadgeClass('UP');
+        const downBadgeClass = comp.getBadgeClass('DOWN');
+        expect(upBadgeClass).toEqual('bg-success');
+        expect(downBadgeClass).toEqual('bg-danger');
     });
 
-    describe('refresh', () => {
-        it('should call refresh on init', () => {
-            // GIVEN
-            const health: Health = { status: 'UP', components: { mail: { status: 'UP', details: { mailDetail: 'mail' } } } };
-            jest.spyOn(service, 'checkHealth').mockReturnValue(of(health));
+    it('should call refresh on init', () => {
+        // GIVEN
+        const health: Health = { status: 'UP', components: { mail: { status: 'UP', details: { mailDetail: 'mail' } } } };
+        jest.spyOn(healthService, 'checkHealth').mockReturnValue(of(health));
 
-            // WHEN
-            comp.ngOnInit();
+        // WHEN
+        comp.ngOnInit();
 
-            // THEN
-            expect(service.checkHealth).toHaveBeenCalled();
-            expect(comp.health).toEqual(health);
-        });
+        // THEN
+        expect(healthService.checkHealth).toHaveBeenCalledTimes(1);
+        expect(comp.health).toBe(health);
```

Comment:
Is there a particular reason why you switched from `toEqual()` to `toBe()`? Since health is an object, I would suggest using `toEqual()`.
```suggestion
        expect(comp.health).toEqual(health);
```

Reply:
added in e94431d